pkg/apiserver/certificate/certificate.go

```diff
@@ -110,5 +107,94 @@ func ApplyServerCert(selfSignedCert bool, client kubernetes.Interface, aggregato
 	}
 
 	caCertController := newCACertController(caContentProvider, client, aggregatorClient)
+
+	if selfSignedCert {
+		rotationDuration, err := nextRotationDuration(secureServing, maxRotateDuration)
+		if err != nil {
+			return nil, fmt.Errorf("error reading expiration date of cert: %v", err)
+		}
+
+		time.AfterFunc(rotationDuration, func() {
+			RotateSelfSignedCertificates(caCertController, secureServing)
+		})
```

Comment:
Perhaps it can be simpler like the following to avoid a few redundant code.
```
go func() {
	for {
		rotationDuration, err := nextRotationDuration(secureServing, maxRotateDuration)
		if err != nil {
			klog.Errorf("error reading expiration date of cert: %v", err)
			return
		}
		time.Sleep(rotationDuration)

		caContentProvider, err := generateSelfSignedCertificate(secureServing)
		if err != nil {
			klog.Errorf("error generating new cert: %v", err)
			return
		}
		caCertController.UpdateCertificate(caContentProvider)
	}
}()
```